nomad/structs/structs.go

```diff
@@ -10273,6 +10279,40 @@ func (a *Allocation) Reconnected() (bool, bool) {
 	return true, a.Expired(lastReconnect)
 }
 
+func (a *Allocation) ToIdentityClaims() *IdentityClaims {
+	now := jwt.NewNumericDate(time.Now().UTC())
+	return &IdentityClaims{
+		Namespace:    a.Namespace,
+		AllocationID: a.ID,
+		RegisteredClaims: jwt.RegisteredClaims{
+			// TODO: in Nomad 1.5.0 we'll have a refresh loop to
+			// prevent allocation identities from expiring before the
+			// allocation is terminal. Once that's implemented, add an
+			// ExpiresAt here ExpiresAt: &jwt.NumericDate{},
```

Comment:
This is still a maybe to me as JWTs can be fully stateless without expiration or revocation:

If the signature is signed by a valid key and the alloc is alive, the jwt is valid!

If a jwt gets leaked, kill the alloc, you can't trust its identity!

Although without expiration no one can do offline attenuation: you have to hit a Nomad API to validate the JWT. If we're ok with tokens living _past_ an allocation's termination, then expiration means any valid Nomad JWT is valid without having to ask Nomad about it.

I don't know though. Definitely a critical thing to think through.

Reply:
> Although without expiration no one can do offline attenuation: you have to hit a Nomad API to validate the JWT. If we're ok with tokens living past an allocation's termination, then expiration means any valid Nomad JWT is valid without having to ask Nomad about it.

Yeah my thinking here is that one of our big internally-proposed use cases is for Vault, and being able to get just the public key from Nomad and use it for many token validations would cut down on cross-traffic.